lib/mix/lib/mix.ex

```diff
@@ -1013,6 +1056,21 @@ defmodule Mix do
     end
   end
 
+  @doc """
+  Returns the directory where the current `Mix.install/2` project
+  resides.
+  """
+  @spec install_project_dir() :: Path.t()
+  def install_project_dir() do
```

Comment:
Nitpick: let's just call it `install_dir`.

Reply:
The reason I picked more explicit name is because we already have `MIX_INSTALL_DIR`, so install dir means the top-level location. If we start mixing both, it may be confusing?

Reply:
Agreed with your rationale. :)